### PR TITLE
Complete rework of the slider

### PR DESCRIPTION
**Pull request by BFH-ktt1**
The maximum number of bits (slider size) can be selected on top. For the moment I put 8 (256 steps). Addresses #988.

### CODE REVIEW

**Review by maehne — DISMISSED** [review]
LGTM

File: src/main/java/com/cburch/logisim/std/io/extra/Slider.java [hunk 1]
```diff
@@ -65,50 +64,50 @@
...
     }
   }
 
   public static class SliderValue implements InstanceData, Cloneable {
-    private long currentvalue = 0;
-    private byte bitwidth = 8, currentx = 0;
-    private boolean right_to_left = false;
+    private int nrOfBits = MAXIMUM_NUMBER_OF_BITS;
+    private int sliderPosition = 0;
+    private boolean rightToLeft = false;
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/io/extra/Slider.java:110]: `var` instead of `boolean` please.

File: src/main/java/com/cburch/logisim/std/io/extra/Slider.java [hunk 2]
```diff
@@ -118,46 +117,29 @@ public Object clone() {
...
-      return this.currentvalue;
+    
+    public int getCurrentValue() {
+      final var completeValue = rightToLeft ? MAXIMUM_SLIDER_POSITION - sliderPosition : sliderPosition; 
+      return completeValue >> (MAXIMUM_NUMBER_OF_BITS - nrOfBits);
     }
-
-    public byte getCurrentX() {
-      return this.currentx;
+
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/io/extra/Slider.java:125]: Diff shows this is empty indentation in this line. If so, please remove.

File: src/main/java/com/cburch/logisim/std/io/extra/Slider.java [hunk 3]
```diff
@@ -118,46 +117,29 @@ public Object clone() {
         return null;
       }
     }
-
-    public long getCurrentValue() {
-      return this.currentvalue;
+    
+    public int getCurrentValue() {
+      final var completeValue = rightToLeft ? MAXIMUM_SLIDER_POSITION - sliderPosition : sliderPosition;
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/io/extra/Slider.java:122]: Can we have `(MAXIMUM_SLIDER_POSITION - sliderPosition)` (in brackets) for better readability?

File: src/main/java/com/cburch/logisim/std/io/extra/Slider.java [hunk 4]
```diff
@@ -118,46 +117,29 @@ public Object clone() {
...
+      return sliderPosition;
     }
 
-    public void setCurrentBitWidth(int b) {
-      if (b != this.bitwidth) {
-        this.bitwidth = (byte) b;
-        setCurrentX();
-      }
+    public void setSliderPosition(int value) {
+      sliderPosition = (value < 0) ? 0 : (value > MAXIMUM_SLIDER_POSITION) ? MAXIMUM_SLIDER_POSITION : value;
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/io/extra/Slider.java:131]: I think you can shorten this to

    sliderPosition = Math.max(0, Math.min(value, MAXIMUM_SLIDER_POSITION));

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/std/io/extra/Slider.java (hunk 1 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/io/extra/Slider.java:110]: Sorry, but as these are global class variables, it won't accept var. I could make an initialiser where they are assigned, but that would provide more LOC

Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/std/io/extra/Slider.java:110]: Ok, no need to touch then.

**Review by MarcinOrlowski — APPROVED** [review]

**Comment by MarcinOrlowski** [comment]
LGTM
